plot/plot.app.js

```diff
@@ -174,16 +172,18 @@
                       return layout;
                   }
                 }
-
-                function formatData(data) { // HACK: for the demo data
-                  var formated_data = parseInt(data.split(' ')[0], 10);
-                  if (isNaN(formated_data)) {
-                    return data
+                function formatData(data) {
+                  try {
+                    var formated_data = parseInt(data.split(' ')[0], 10);
```

Comment:
This fix still allows string data where spaces and integers are present to be arbitrarily reformatted, which should never be the default behavior. The execution of this function needs to be configurable at the plot level.